pkg/event/event.go

```diff
@@ -205,9 +277,17 @@ func (e Event) InngestMetadata() (*InngestMetadata, error) {
 	}
 }
 
-func NewOSSTrackedEvent(e Event) TrackedEvent {
+func NewOSSTrackedEvent(e Event, seed *SeededID) TrackedEvent {
 	// Never use e.ID as the internal ID, since it's specified by the sender
 	internalID := ulid.MustNew(ulid.Now(), rand.Reader)
+
+	if seed != nil {
+		newInternalID, err := seed.ToULID()
+		if err == nil {
+			internalID = newInternalID
+		}
+	}
+
 	if e.ID == "" {
 		e.ID = internalID.String()
 	}
```

Comment:
Shouldn't we use this to only overwrite the `ID` and not touch the `internalID`? `ossTrackedEvent.Id` should always be ours and protected.
```suggestion
	if seed != nil && e.ID == "" {
		newIdempotencyKey, err := seed.ToULID()
		if err == nil {
			e.ID = newIdempotencyKey.String()
		}
	}
```

Reply:
No. Events are not filtered at the point of ingestion - see inngest/monorepo#2682.